### PR TITLE
listen1: init at 2.31.0

### PR DESCRIPTION
**Pull request by running-grass**
## Description of changes

Add https://github.com/listen1/listen1_desktop

one for all free music in china (Windows, Mac, Linux desktop)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by running-grass** [comment]
Result of `nixpkgs-review pr 259496` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>listen1</li>
  </ul>
</details>

**Comment by running-grass** [comment]
@NickCao hi, could you review and merge this pr ?

File: pkgs/applications/audio/listen1/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  pname = "listen1";
+  version = "2.31.0";
+  name = "listen1-${version}";
+  nameExecutable = pname;
+  src = fetchurl {
+    url = "https://github.com/listen1/listen1_desktop/releases/download/v${version}/listen1_${version}_linux_x86_64.AppImage";
+    name = "listen1-${version}.AppImage";
+    sha256 = "9d80ca7b157356ec0f9affde2bd701d280128171196eb3ebcea3eee4e1e4e8d4";
+  };
+  appimageContents = appimageTools.extractType2 { inherit name src; };
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:13]:
```suggestion
  appimageContents = appimageTools.extractType2 { inherit pname version src; };
```

File: pkgs/applications/audio/listen1/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ lib, fetchurl, appimageTools, makeWrapper }:
+
+let
+  pname = "listen1";
+  version = "2.31.0";
+  name = "listen1-${version}";
+  nameExecutable = pname;
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:7]:
```suggestion
```

File: pkgs/applications/audio/listen1/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  src = fetchurl {
+    url = "https://github.com/listen1/listen1_desktop/releases/download/v${version}/listen1_${version}_linux_x86_64.AppImage";
+    name = "listen1-${version}.AppImage";
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:10]: Any reason for the specific name?

File: pkgs/applications/audio/listen1/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    sha256 = "9d80ca7b157356ec0f9affde2bd701d280128171196eb3ebcea3eee4e1e4e8d4";
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:11]:
```suggestion
    hash = "...";
```

Please use SRI hash format.

File: pkgs/applications/audio/listen1/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  };
+  appimageContents = appimageTools.extractType2 { inherit name src; };
+in
+appimageTools.wrapType2 {
+  inherit name src;
+
+  extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs)
+    ++ [ pkgs.libsecret ];
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:19]: Or just?

```suggestion
  extraPkgs = pkgs: [ pkgs.libsecret ];
```

File: pkgs/applications/audio/listen1/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  nameExecutable = pname;
+  src = fetchurl {
+    url = "https://github.com/listen1/listen1_desktop/releases/download/v${version}/listen1_${version}_linux_x86_64.AppImage";
+    name = "listen1-${version}.AppImage";
+    sha256 = "9d80ca7b157356ec0f9affde2bd701d280128171196eb3ebcea3eee4e1e4e8d4";
+  };
+  appimageContents = appimageTools.extractType2 { inherit name src; };
+in
+appimageTools.wrapType2 {
+  inherit name src;
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:16]:
```suggestion
  inherit pname version src;
```

File: pkgs/applications/audio/listen1/default.nix [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  extraPkgs = pkgs: (appimageTools.defaultFhsEnvArgs.multiPkgs pkgs)
+    ++ [ pkgs.libsecret ];
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    source "${makeWrapper}/nix-support/setup-hook"
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:23]: Adding `makeWrapper` to `nativeBuildInputs` should be better.

File: pkgs/applications/audio/listen1/default.nix [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    install -m 444 -D ${appimageContents}/usr/share/icons/hicolor/512x512/apps/listen1.png \
+      $out/share/icons/hicolor/512x512/apps/listen1.png
+  '';
+
+  meta = with lib; {
+    description = "One for all free music in China";
+    homepage = "http://listen1.github.io/listen1/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ running-grass ];
+    platforms = [ "x86_64-linux" ];
```
Comment (NickCao) [inline pkgs/applications/audio/listen1/default.nix:38]: Also set `sourceProvenance` and `mainProgram`

File: pkgs/top-level/all-packages.nix [hunk 9]
```diff
@@ -33454,6 +33454,8 @@ with pkgs;
 
   linuxsampler = callPackage ../applications/audio/linuxsampler { };
 
+  listen1 = callPackage ../applications/audio/listen1 { };
```
Comment (NickCao) [inline pkgs/top-level/all-packages.nix:33457]: Please use the `pkgs/by-name` hierarchy, read `pkgs/by-name/README.md` for instructions.

File: pkgs/by-name/li/listen1/package.nix [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  '';
+
+  meta = with lib; {
+    description = "One for all free music in China";
+    homepage = "http://listen1.github.io/listen1/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ running-grass ];
+    platforms = [ "x86_64-linux" ];
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    mainProgram = "Listen1";
```
Comment (NickCao) [inline pkgs/by-name/li/listen1/package.nix:31]:
```suggestion
    mainProgram = "listen1";
```

**Review by NickCao — COMMENTED** [review]
Apart from `mainProgram`, LGTM. Tested basic functionality, notice this version is still affected by https://github.com/listen1/listen1_desktop/issues/1132.

File: pkgs/by-name/li/listen1/package.nix [hunk 11]
```diff
@@ -0,0 +1,33 @@
+{ lib, fetchurl, appimageTools, makeWrapper }:
```
Comment (NickCao) [inline pkgs/by-name/li/listen1/package.nix:1]:
```suggestion
{ lib, fetchurl, appimageTools }:
```

**Review by NickCao — APPROVED** [review]

**Comment by running-grass** [comment]
@NickCao Thank you for your patient guidance

**Pull request merged**